src/components/structures/MessagePanel.tsx

```diff
@@ -863,6 +864,18 @@ export default class MessagePanel extends React.Component<IProps, IState> {
         return receipts;
     }
 
+    public getReadReceiptGroupForEvents(mxEvents: MatrixEvent[]): ReactNode {
+        const lastEvent = mxEvents ? mxEvents[mxEvents.length - 1] : null;
+        const totalReceipts = lastEvent && this.readReceiptsByEvent[lastEvent.event.event_id];
+
+        return <ReadReceiptGroup
+            readReceipts={totalReceipts || []}
+            readReceiptMap={this.readReceiptMap}
+            checkUnmounting={this.isUnmounting}
+            isTwelveHour={this.props.isTwelveHour}
+        />;
```

Comment:
this seems to only show the RR for the last event? isn't that more confusing that the union?

Reply:
If we've got messages A B and C in a group, and a user acknowledges message C, then we know the user has read all of the messages.

Using the RRs of the last event ensures that we only show RRs for the group if a user has acknowledged all the messages in the group. The union would also show RRs of users who had only read some of the messages in the group

Reply:
Yes but imagine events:

Message 1
Message 2
State 1
State 2
State 3

(State events being in a group)

If a user's RR is set on a Message (1/2) and then they send a read receipt for State (1/2) then their RR will vanish, which will look broken IMO

Reply:
That vanishing is already what happens today. But if we show a union of read receipts, we’d risk the user not understanding what has been read and what hasn’t.

Should product take a look at this and make the decision, as nique suggested last week?